src/installer.ts

```diff
@@ -2,187 +2,156 @@
 import * as core from '@actions/core';
 import * as exec from '@actions/exec';
 import * as io from '@actions/io';
-import hc = require('@actions/http-client');
+import * as hc from '@actions/http-client';
 import {chmodSync} from 'fs';
-import * as path from 'path';
-import {ExecOptions} from '@actions/exec/lib/interfaces';
-import * as semver from 'semver';
-
-const IS_WINDOWS = process.platform === 'win32';
-
-/**
- * Represents the inputted version information
- */
-export class DotNetVersionInfo {
-  public inputVersion: string;
-  private fullversion: string;
-  private isExactVersionSet: boolean = false;
-
-  constructor(version: string) {
-    this.inputVersion = version;
-
-    // Check for exact match
-    if (semver.valid(semver.clean(version) || '') != null) {
-      this.fullversion = semver.clean(version) as string;
-      this.isExactVersionSet = true;
-
-      return;
-    }
-
-    const parts: string[] = version.split('.');
-
-    if (parts.length < 2 || parts.length > 3) this.throwInvalidVersionFormat();
-
-    if (parts.length == 3 && parts[2] !== 'x' && parts[2] !== '*') {
-      this.throwInvalidVersionFormat();
-    }
+import path from 'path';
+import semver from 'semver';
+import {IS_LINUX, IS_WINDOWS, logWarning} from './utils';
+
+export interface IDotNetVersion {

```

Comment:
```suggestion
export interface DotNetVersion {
```

TypeScript doesn't conventionally use the `I-` prefix on interfaces.